lib/utils/request.js

```diff
@@ -207,6 +208,26 @@ function MULTIGET (fhurls, where, cb) {
 }
 
 /**
+ * Creates a closure over a callback that handles error and status code checking for a upload request
+ * @param  {Function} cb Callback function
+ * @return {Function}      Node-style callback closed over `cb`
+ * @access private
+ */
+function uploadHandlerFor(cb) {
+  return function(err, response) {
+    if(err) {
+      return cb("Error Uploading File " + err);
+    }
+
+    if (response.statusCode.toString()[0]!=='2') {
+      return cb("Error - non 2xx status code: " + response.statusCode + " " + response.body);
```

Comment:
Avoid passing response body because may contain some binary content. We should agree how api should be have. If status is 200 you can expect to have binary octet stream (zip file). Maybe json otherwise with body.userDetail as used in other places?